### PR TITLE
[chore] Fix copyright headers & fixed issue with mypy & NumPy versions in pre-commit

### PR DESCRIPTION
**Pull request by min-xu-ai**
## What does this PR do?
Fixes an internal task T70826226.

## Before submitting

- [ ] Did you have fun?
  - Make sure you had fun coding 🙃
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/main/CONTRIBUTING.md)?
- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
  - [ ] N/A
- [ ] Did you make sure to update the docs?
  - [ ] N/A
- [ ] Did you write any new necessary tests?
  - [ ] N/A
- [ ] Did you update the [changelog](https://github.com/facebookresearch/fairscale/blob/main/CHANGELOG.md)? (if needed)
  - [ ] N/A

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

**Comment by tmarkstrum** [comment]
I think we should replace Facebook with Meta.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,3 +1,8 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
```
Comment (tmarkstrum) [inline .pre-commit-config.yaml:1]: Meta?

**Comment by min-xu-ai** [comment]
> I think we should replace Facebook with Meta.

good point. I think that should be done with all files. For this PR, I think we should just use Facebook to be consistent and then later change all files in one PR. What do you think?

**Review by tmarkstrum — APPROVED** [review]
LGTM. I will create a issue to change Facebook to Meta in the code base.

File: pyproject.toml [hunk 2]
```diff
@@ -27,4 +27,4 @@ use_parentheses = true
 skip_glob = ["build/*", "stubs/*"]
 # Don't split "import" and "from".
 force_sort_within_sections = true
-known_third_party = ["benchmark_dataset", "datasets", "golden_configs", "models", "numpy", "parameterized", "pytest", "recommonmark", "setuptools", "sklearn", "torch", "torchtext", "torchvision", "utils"]
```
Comment (anj-s) [inline pyproject.toml:30]: Does this fix the mypy issue that we keep having?

**Review by anj-s — APPROVED** [review]

File: pyproject.toml (hunk 2 above)
Comment (min-xu-ai) [reply pyproject.toml:30]: no, it doesn't. I think the way pre-commit handles local tools is causing this for me. I am trying to get my local env to behave like the GitHub actions.

**Pull request merged**